splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+
+class AppStartupTimer {
+    private final AtomicLong firstPossibleTimestamp = new AtomicLong(System.currentTimeMillis());
+    private final AtomicReference<Span> overallAppStartSpan = new AtomicReference<>();
```

Comment:
Nit: it probably could be just a `volatile` field, you're not using CAS anywhere in this class.

Reply:
TBH, I don't know how volatile works on Android, and I tend to favor atomics over bare volatiles, but I'll do a bit of research on whether volatile has the same semantics as on the standard JVM.

Reply:
cool. looks like as of Android 4, it does work the same as it is supposed to in the JMM:  https://stackoverflow.com/questions/6973667/dalvik-vm-java-memory-model-concurrent-programming-on-android